src/docfx/lib/github/GitHubAccessor.cs

```diff
@@ -58,87 +58,35 @@ public async Task<(Error, GitHubUser)> GetUserByLogin(string login)
-                        },
-                    },
-                    data = new
-                    {
-                        user = new
-                        {
-                            name = "",
-                            email = "",
-                            login = "",
-                            databaseId = 0,
-                        },
-                    },
-                });
-
-                if (grahpApiResponse.errors != null && grahpApiResponse.errors.Length != 0)
-                {
-                    var notFoundError = grahpApiResponse.errors.FirstOrDefault(e => e.type == "NOT_FOUND" && e.path.Contains("user"));
-                    if (notFoundError != null)
-                        return default;
-
-                    var rateLimitError = grahpApiResponse.errors.FirstOrDefault(e => e.type == "MAX_NODE_LIMIT_EXCEEDED" || e.type == "RATE_LIMITED");
-                    if (rateLimitError != null)
-                    {
-                        _rateLimitError = Errors.GitHubApiFailed(rateLimitError.message);
-                    }
-
-                    return (Errors.GitHubApiFailed(rateLimitError?.message ?? grahpApiResponse.errors.First().message), null);
-                }
-
-                return (null, new GitHubUser
-                {
-                    Id = grahpApiResponse.data.user.databaseId,
-                    Login = grahpApiResponse.data.user.login,
-                    Name = string.IsNullOrEmpty(grahpApiResponse.data.user.name) ? grahpApiResponse.data.user.login : grahpApiResponse.data.user.name,
-                    Emails = !string.IsNullOrEmpty(grahpApiResponse.data.user.email) ? new[] { grahpApiResponse.data.user.email } : Array.Empty<string>(),
-                });
-            }
-            catch (Exception ex)
+            if (error != null)
             {
-                Log.Write(ex);
-                return (Errors.GitHubApiFailed(ex.InnerException?.Message ?? ex.Message), null);
+                return (error, null);
             }
-        }
 
-        public async Task<(Error, IEnumerable<GitHubUser>)> GetUsersByCommit(string owner, string name, string commit)
-        {
-            if (_rateLimitError != null)
+            return (null, new GitHubUser
             {
-                return default;
-            }
+                Id = data.user.databaseId,
+                Login = data.user.login,
+                Name = string.IsNullOrEmpty(data.user.name) ? data.user.login : data.user.name,
+                Emails = new[] { data.user.email }.Where(email => !string.IsNullOrEmpty(email)).ToArray(),
+            });
+        }
 
-            if (_unauthorizedError != null)
+        public async Task<(Error, IEnumerable<GitHubUser>)> GetUsersByCommit(string owner, string name, string commit, string authorEmail = null)
+        {
+            // Stop calling GitHub whenever any error is returned from github
+            if (_error != null)
```

Comment:
that's too aggressive, one single network issue error doesn't means we can't use github anymore

Reply:
transient errors are covered by retry, the errors here are non-transient errors.

Reply:
we only retry for three times, can't cover all temporary transient errors

Reply:
updated to only cache non-transient errors